### PR TITLE
150 add breeder screen

### PR DESCRIPTION
**Pull request by haakonjacobsen**
Added:

Backend:
- [x] Review schema and review endpoints
- [x] User schema and user endpoints
- [x] Updated breeder schema and endpoints
- [x] Validation logic for creating users and reviews

Frontend:
- [x] New Breeder screen
- [x] A reusable review component
- [x] New SVG's with filled color for verified users and reviews
- [x] Updated theme and styles to be more reusable 

bitmoji

### CODE REVIEW

File: backend_rest/src/api/reviews.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  const posts = await Review.find().sort({ createdAt: -1 });
+  res.status(status.OK).json(posts);
+};
+
+const getReviewByBreederId = async (req: Request, res: Response) => {
+  try {
+    const { id } = req.params;
+    const reviews = await Review.find({ reviewOf: id })
+      .populate([{ path: 'reviewBy', model: User }])
+      .sort({ createdAt: 1 });
```
Comment (AndreasWintherMoen) [inline backend_rest/src/api/reviews.ts:20]: Won't `createdAt: 1` give oldest first?

File: backend_rest/src/api/users.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  res.status(status.OK).json(posts);
+};
+
+const createUser = async (req: Request, res: Response) => {
+  try {
+    const user = req.body;
+    const createdUser = await User.create(user);
+    return res.status(status.CREATED).json(createdUser);
+  } catch (_error: any) {
+    if (_error.code === 11000) {
```
Comment (AndreasWintherMoen) [inline backend_rest/src/api/users.ts:20]: We should have a dict defining MongoDB error codes. 11000 doesn't tell me anything :D Add a TODO about it here and I'll create an issue so we can do it later.

File: frontend/api/user/getBreederReviewsByID.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { IReview } from '../../types/databaseTypes';
+import config from '../../config';
+
+export const getBreederReviewsByID = async (breederID: string) => {
+  const url = `${config.API_BASE_URI}/api/reviews/find-by-breeder/${breederID}`;
+  console.log(url);
```
Comment (AndreasWintherMoen) [inline frontend/api/user/getBreederReviewsByID.ts:6]: Unnecessary console log

File: frontend/components/elements/UserReview.tsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        <View style={defaultStyles.svgContainer40}>
+          <RatingStarFilled rating={review.rating} color={theme.palette.bark.primary} />
+        </View>
+      </View>
+      <Text style={textStyles.body1}>
+        {review.description.length < 400 || expanded
+          ? review.description
+          : getShortDescription(review.description)}
+      </Text>
+      {review.description.length >= 400 ? (
```
Comment (AndreasWintherMoen) [inline frontend/components/elements/UserReview.tsx:47]: `review.description.length >= 400 && (`

...


`)}` <- this means removing ` : null`

File: frontend/screens/BreederScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+  const {
+    data: reviews,
+    refetch: refetchReviews,
+    remove: removeReviews
+  } = useQuery(['getBreederReviewsByID', breederId], () => getBreederReviewsByID(breederId), {
+    enabled: !!breeder
+  });
+
+  const onRefresh = () => {
+    remove();
```
Comment (AndreasWintherMoen) [inline frontend/screens/BreederScreen.tsx:43]: These should be called `removeBreeder` and `refetchBreeder` for consistency.

File: frontend/screens/DogPostScreen.tsx [hunk 6]
```diff
@@ -147,7 +152,13 @@ export default function DogPostScreen({ route, navigation }: DogPostScreenProps)
                 />
               ) : null}
             </View>
-            <TouchableOpacity onPress={() => console.log('navigate to breeder page')}>
+            <TouchableOpacity
+              onPress={() =>
+                navigation.push('BreederScreen', {
+                  breederId: data?.breeder._id
```
Comment (AndreasWintherMoen) [inline frontend/screens/DogPostScreen.tsx:158]: Can data be null? What would happen then? Haven't you checked for falsy data earlier?

File: frontend/styles/textStyles.tsx [hunk 7]
```diff
@@ -59,12 +59,40 @@ export const textStyles = StyleSheet.create({
   },
   searchResultText: {
     fontSize: 14,
-    color: theme.palette.text.secondary,
+    color: theme.palette.text.reallyDarkGray,
```
Comment (AndreasWintherMoen) [inline frontend/styles/textStyles.tsx:62]: Best name :D

**Comment by AndreasWintherMoen** [comment]
bitmoji

File: backend_rest/src/api/reviews.ts (hunk 1 above)
Comment (haakonjacobsen) [reply backend_rest/src/api/reviews.ts:20]: True, it's is better to have the most recent comments first. Will fix.

File: frontend/components/elements/UserReview.tsx (hunk 4 above)
Comment (haakonjacobsen) [reply frontend/components/elements/UserReview.tsx:47]: True, fixed

File: frontend/screens/DogPostScreen.tsx (hunk 6 above)
Comment (haakonjacobsen) [reply frontend/screens/DogPostScreen.tsx:158]: Not sure. We are fetching data from the backend, and we have two if checks for isLoading and isError. Therefore we can assume that the fetching was successful on the last return, but if the return is successful and it returns null, i'm not sure what will happen.

**Review by AndreasWintherMoen — APPROVED** [review]

**Comment by AndreasWintherMoen** [comment]
bitmoji

**Pull request merged**